spec/rubocop/cop/rails/action_name_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Rails::ActionName, :config do
+  context 'when non configured name is used for private method' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        class UsersController < ApplicationController
+          private
+
+          def articles
+          end
+        end
+      RUBY
+    end
+  end
+
+  context 'when configured name is used for public method' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        class UsersController < ApplicationController
+          def index
+          end
+        end
+      RUBY
+    end
+  end
+
+  context 'when manually configured name is used for public method' do
+    let(:cop_config) do
+      { 'ActionNames' => %w[articles] }
+    end
+
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        class UsersController < ApplicationController
+          def articles
+          end
+        end
+      RUBY
+    end
+  end
+
+  context 'when non configured name is used for public method' do
+    it 'registers an offense' do
+      expect_offense(<<~RUBY)
+        class UsersController < ApplicationController
+          def articles
+              ^^^^^^^^ Use only specific action names.
```

Comment:
Perhaps the message should include the list of configured actions, e.g.:

`Use only specific action names: index, create, show...`